src/libs/OptionsListUtils.js

```diff
@@ -56,19 +56,22 @@ function getPersonalDetailsForLogins(logins, personalDetails) {
  * @return {String}
  */
 function getSearchText(report, personalDetailList) {
-    const searchTerms = [];
+    // Use a set because it skips adding duplicates,
+    // which could potentially significantly speed up our regex search if someone has thousands of chats
+    const searchTerms = new Set();
 
     _.each(personalDetailList, (personalDetail) => {
-        searchTerms.push(personalDetail.displayName);
-        searchTerms.push(personalDetail.login);
+        searchTerms.add(personalDetail.displayName);
+        searchTerms.add(personalDetail.login);
     });
 
     if (report) {
-        searchTerms.push(...report.reportName.split(',').map(name => name.trim()));
-        searchTerms.push(...report.participants);
+        searchTerms.add(report.reportName);
+        searchTerms.add(...report.reportName.split(',').map(name => name.trim()));
+        searchTerms.add(...report.participants);
     }
 
-    return _.unique(searchTerms).join(' ');
+    return _.unique(Array.from(searchTerms)).join(' ');
```

Comment:
NAB: since you're returning an array here, but why did you stick with set and convert it to an array, rather than just start with an array?

Reply:
+1 especially since we are also removing duplicates with `_.unique()`. 
`Set` seems like it would be more performant, but not sure I could say by how much.

Reply:
True, we can remove `_.unique()` if I'm using a `Set` here. 

> Set seems like it would be more performant, but not sure I could say by how much.

`Set` allows us to check whether it has a value much faster than an array. An array requires iterating through each member, a Set can directly look up a value you provide it and tell you if it exists or not. If we had a `Set` vs. an `Array` with 10 values, the `Set` can be up to 10 times faster than the `Array`.

Reply:
I'll get rid of the set usage!